test/sqllogictest/not-null-propagation.slt

```diff
@@ -412,14 +412,19 @@ Explained Query:
                 Filter (#1 = 1) // { types: "(integer?, integer, integer, boolean)" }
                   Map (1, true) // { types: "(integer?, integer, integer, boolean)" }
                     Get l0 // { types: "(integer?, integer)" }
-              Map (false) // { types: "(integer?, integer, boolean)" }
-                Union // { types: "(integer?, integer)" }
-                  Map (1) // { types: "(integer?, integer)" }
-                    Negate // { types: "(integer?)" }
-                      Project (#0) // { types: "(integer?)" }
-                        Filter (#1 = 1) // { types: "(integer?, integer)" }
-                          Get l0 // { types: "(integer?, integer)" }
-                  Get l0 // { types: "(integer?, integer)" }
+              Project (#0, #1, #4) // { types: "(integer?, integer, boolean)" }
+                Map (false) // { types: "(integer?, integer, integer?, integer, boolean)" }
+                  Join on=(#0 = #2 AND #1 = #3) type=differential // { types: "(integer?, integer, integer?, integer)" }
+                    ArrangeBy keys=[[#0, #1]] // { types: "(integer?, integer)" }
+                      Union // { types: "(integer?, integer)" }
+                        Map (1) // { types: "(integer?, integer)" }
+                          Negate // { types: "(integer?)" }
+                            Project (#0) // { types: "(integer?)" }
+                              Filter (#1 = 1) // { types: "(integer?, integer)" }
+                                Get l0 // { types: "(integer?, integer)" }
+                        Get l0 // { types: "(integer?, integer)" }
+                    ArrangeBy keys=[[#0, #1]] // { types: "(integer?, integer)" }
+                      Get l0 // { types: "(integer?, integer)" }
```

Comment:
I can see a diff in main with a simpler version of the query:

```sql
SELECT
  1 = SOME (VALUES(col_null)), 
  1 = SOME (VALUES(col_not_null))
FROM
  int_table;
```

Reply:
Checking back in on this: since landing https://github.com/MaterializeInc/materialize/pull/25551 I don't see a big difference between these (this PR version is shorter, inheriting some of the simplifications from that other PR). That PR was meant to work around some regressions, so perhaps they are addressed?

Reply:
The selected diff has been removed as of the above PR too. Will mark as resolved, but please open back up if I've missed the point!